windows-installer/construct.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+import ctypes
+import subprocess
+import os
+import sys
+import shutil
+
+
+def main():
+    print('Gather runtime data')
+
+    try:
+        subprocess.check_output(['choco', '--version'])
+    except subprocess.CalledProcessError:
+        raise RuntimeError('Error: Chocolatey (https://chocolatey.org/) needs'
+                           'to be installed to run this script.')
+
+    script_path = os.path.realpath(__file__)
+    repo_path = os.path.dirname(os.path.dirname(script_path))
+    build_path = os.path.join(repo_path, 'windows-installer', 'build')
+
+    venv_path = os.path.join(build_path, 'venv-config')
+    venv_python = os.path.join(venv_path, 'Scripts', 'python.exe')
+    installer_cfg_path = os.path.join(build_path, 'installer.cfg')
+    wheels_path = os.path.join(build_path, 'wheels')
+
+    certbot_version = subprocess.check_output([sys.executable, '-c', 'import certbot; print(certbot.__version__)'],
+                                              universal_newlines=True, cwd=repo_path).strip()
+
+    certbot_packages = ['acme', '.']
+    certbot_packages.extend([name for name in os.listdir(repo_path) if name.startswith('certbot-dns-')])
+
+    print('Copy assets')
+
+    os.makedirs(build_path, exist_ok=True)
+    shutil.copy(os.path.join(repo_path, 'windows-installer', 'certbot.ico'), build_path)
+    shutil.copy(os.path.join(repo_path, 'windows-installer', 'run.py'), build_path)
+
+    print('Prepare pynsist config')
+
+    with open(os.path.join(installer_cfg_path), 'w') as file_h:
+        file_h.write("""\
+[Application]
+name=Certbot
+version={certbot_version}
+icon=certbot.ico
+publisher=Electronic Frontier Fundation
+script=run.py
+
+[Build]
+directory=nsis
+installer_name=certbot-{certbot_version}-win32_install.exe
+
+[Python]
+version=3.7.0
+
```

Comment:
Should we explicitly specify we want 32 bit here? Without doing so, https://pynsist.readthedocs.io/en/latest/cfgfile.html#python-section makes it sound like the bitness will depend on the Windows system this script is run on.

Reply:
Yes, most of applications on Windows are 32 bits first, and optionally 64 bits.

Done.

Reply:
And... It is not working. Since all systems are 64 bits now, `pip wheel`, that I use in my approach, build for the current Python environment. And Python for Windows is in fact mainly distributed and installed on its 64 bits version.

It will work for the majority of the cases, at least for a beta phase. We can in the future build both 32 bits and 64 bits if we take advantage of the capability of choosing the bitness with pynsist.

This will require however to pass the list of requirements directly to pynsist. This can be associated with the pinning of wheel versions.

I think we can put that in a future PR.

Reply:
You will get an error, while pynsist is collecting the wheels, like this:
`ValueError: Wheel cffi-1.12.3-cp37-cp37m-win_amd64.whl is not compatible with Python 3.7.4, win32`

It depends entirely on the Python version you use, since on Windows you can get 32bits or 64bits flavors.

Ok, I change my mind here: 32bits is still the most conservative choice here, and AppVeyor has Python 3.7 32bits. I move to that.

But we will definitely need to improve that and use the pynsist capability to get wheels independantly from the actual version of Python that runs it.